src/query.rs

```diff
@@ -233,18 +233,40 @@ mod tests {
                 let indexed_crate_new = VersionedIndexedCrate::new(&crate_new);
                 let indexed_crate_old = VersionedIndexedCrate::new(&crate_old);
 
-                let adapter =
-                    VersionedRustdocAdapter::new(&indexed_crate_new, Some(&indexed_crate_old))
-                        .expect("could not create adapter");
-                let results_iter = adapter
-                    .run_query(&semver_query.query, semver_query.arguments.clone())
-                    .unwrap();
-                (
-                    format!("./test_crates/{}/", crate_pair),
-                    results_iter
-                        .map(|res| res.into_iter().map(|(k, v)| (k.to_string(), v)).collect())
-                        .collect::<Vec<BTreeMap<_, _>>>(),
-                )
+                let run_query_on =
+                    |indexed_crate_1: &VersionedIndexedCrate,
+                     indexed_crate_2: &VersionedIndexedCrate| {
+                        let adapter =
+                            VersionedRustdocAdapter::new(indexed_crate_1, Some(indexed_crate_2))
+                                .expect("could not create adapter");
+                        let results_iter = adapter
+                            .run_query(&semver_query.query, semver_query.arguments.clone())
+                            .unwrap();
+                        (
+                            format!("./test_crates/{}/", crate_pair),
+                            results_iter
+                                .map(|res| {
+                                    res.into_iter().map(|(k, v)| (k.to_string(), v)).collect()
+                                })
+                                .collect::<Vec<BTreeMap<_, _>>>(),
+                        )
+                    };
+
+                assert!(
+                    run_query_on(&indexed_crate_new, &indexed_crate_new)
+                        .1
+                        .is_empty(),
+                    "{}",
+                    crate_pair
+                );
+                assert!(
+                    run_query_on(&indexed_crate_old, &indexed_crate_old)
+                        .1
+                        .is_empty(),
+                    "{}",
+                    crate_pair
+                );
+                run_query_on(&indexed_crate_new, &indexed_crate_old)
```

Comment:
Would you mind adding some comments about what's going on here and why? Otherwise this code seems pretty unusual and confusing...

Reply:
Isn't it self-explanatory? In short, I'm asserting that running the crate compared to the same crate returns no errors.

Reply:
What do you think about the new version? @obi1kenobi

Reply:
Fixed in b58de954ac788d4a4eab30f13eba324cb670d4ee